kodi_addon_checker/schema_validation.py

```diff
@@ -0,0 +1,88 @@
+import os
+import logging
+import xmlschema
+
+from .report import Report
+from .record import PROBLEM, Record, INFORMATION
+
+XML_SCHEMA = os.path.join('.', 'xml_schema')
+LOGGER = logging.getLogger(__name__)
+
+
+def schemas(report: Report, parsed_xml):
+    """validates XML file with existing schemas
+    :parsed_xml: parsed data from an XML file
+    """
+
+    failed, metadatacount, validcount = _validation_checks(report, parsed_xml)
+
+    root_schema = os.path.join(XML_SCHEMA, 'addon.xsd')
+    if not _validate(parsed_xml, root_schema):
+        report.add(Record(PROBLEM, "Schema validation failed for root addon element"))
+        validcount -= 1
+
+    if metadatacount != 1:
+        report.add(Record(PROBLEM, 'Metadata missing/occurred more than once'))
+
+    if validcount == 1:
+        report.add(Record(INFORMATION, "Valid XML file found"))
+    else:
```

Comment:
log only if failed is not empty

Reply:
DO you mean to say I should remove `validcount` ?

Reply:
No, I mean instead of the else do a `elif failed:` so you only report if actually have a failed list.